vendor/assets/javascripts/github_buttons.js

```diff
@@ -11,136 +11,170 @@
+        counter.textContent = obj.data.followers && addCommas(obj.data.followers);
+        counter.setAttribute('aria-label', counter.textContent + ' followers' + LABEL_SUFFIX);
         break;
     }
 
-    // Show the count if asked
-    if (count === 'true' && counter.innerHTML !== 'undefined') {
+    // Show the count if asked and if it's not empty
+    if (count === 'true' && counter.textContent !== '') {
       counter.style.display = 'block';
+      counter.removeAttribute('aria-hidden');
     }
-  }
+  };
 
   // Set href to be URL for repo
-  button.href = 'https://github.com/' + user + '/' + repo + '/';
+  button.href = REPO_URL;
+
+  var title;
 
   // Add the class, change the text label, set count link href
   switch (type) {
     case 'watch':
       if (v === '2') {
         mainButton.className += ' github-watchers';
-        text.innerHTML = 'Watch';
-        counter.href = 'https://github.com/' + user + '/' + repo + '/watchers';
+        text.textContent = 'Watch';
+        counter.href = REPO_URL + '/watchers';
       } else {
         mainButton.className += ' github-stargazers';
-        text.innerHTML = 'Star';
-        counter.href = 'https://github.com/' + user + '/' + repo + '/stargazers';
+        text.textContent = 'Star';
+        counter.href = REPO_URL + '/stargazers';
       }
+
+      title = text.textContent + ' ' + USER_REPO;
       break;
     case 'star':
       mainButton.className += ' github-stargazers';
-      text.innerHTML = 'Star';
-      counter.href = 'https://github.com/' + user + '/' + repo + '/stargazers';
+      text.textContent = 'Star';
+      counter.href = REPO_URL + '/stargazers';
+      title = text.textContent + ' ' + USER_REPO;
       break;
     case 'fork':
       mainButton.className += ' github-forks';
-      text.innerHTML = 'Fork';
-      button.href = 'https://github.com/' + user + '/' + repo + '/fork';
-      counter.href = 'https://github.com/' + user + '/' + repo + '/network';
+      text.textContent = 'Fork';
+      button.href = REPO_URL + '/fork';
+      counter.href = REPO_URL + '/network';
+      title = text.textContent + ' ' + USER_REPO;
       break;
     case 'follow':
       mainButton.className += ' github-me';
-      text.innerHTML = 'Follow @' + user;
-      button.href = 'https://github.com/' + user;
-      counter.href = 'https://github.com/' + user + '/followers';
+      text.textContent = 'Follow @' + user;
+      button.href = GITHUB_URL + user;
+      counter.href = GITHUB_URL + user + '?tab=followers';
+      title = text.textContent;
+      break;
+    case 'sponsor':
+      mainButton.className += ' github-me';
+      text.textContent = 'Sponsor @' + user;
+      button.href = GITHUB_URL + 'sponsors/' + user;
+      title = text.textContent;
       break;
   }
-  button.setAttribute('aria-label', text.innerHTML + labelSuffix);
 
-  // Change the size
+  button.setAttribute('aria-label', title + LABEL_SUFFIX);
+  document.title = title + LABEL_SUFFIX;
```

Comment:
I think this may be setting the full page's title

Reply:
Actually yes, it does. Original repo is using iframe to render buttons, but we opted out at https://github.com/rubygems/rubygems.org/pull/1827. Sadly we can't just blindly copy upstream (https://github.com/mdo/github-buttons/blob/master/src/js.js) file anymore. I'll open fix.

Reply:
fixed at https://github.com/rubygems/rubygems.org/pull/3143